json-rpc/docs/client_checklist.md

```diff
@@ -0,0 +1,86 @@
+
+# Basics
+
+- [ ] module structure:
+  - libra
+    - LibraClient: high level APIs interface, should support application to do easy mock / stub development.
+    - jsonrpc: jsonrpc client interface, include plain data classes / structs defined in Libra JSON-RPC SPEC document.
+      - types: data transfer object types for jsonrpc client, should match server side JSON-RPC spec data types.
+    - stdlib: move stdlib script utils.
+    - testnet: testnet utils, should include FaucetService for handing testnet mint.
+    - types: Libra onchain data structure types.
+    - utils: includes crypto, data types converting and other utils functions
+      - signing, sha3 hashing, address parsing and converting, hex encoding / decoding
+      - LCS utils
+- [ ] JSON-RPC 2.0 Spec:
+  - spec version validation.
+  - batch requests and responses handling.
+- [ ] JSON-RPC client error handling should distinguish the following 3 type errors:
+  - Transport layer error, e.g. HTTP call failure.
+  - JSON-RPC protocol error: e.g. server respond non json data, or can't be parsed into [Libra JSON-RPC SPEC][1] defined data structure, or missing result & error field.
+  - JSON-RPC error: error returned from server.
+- [ ] https
+- [ ] Client connection pool: keep connection alive for less likely getting inconsistent data from connecting to multiple servers.
+- [ ] Handle stale responses:
+  - [ ] client tracks latest server respond block version and timestamp, raise error when received server response contains stale version / timestamp.
```

Comment:
adding: parse and use libra_chain_id, libra_ledger_version and libra_ledger_tiemstamp in the JSONRPC response